src/ResourceValidator.js

```diff
@@ -1,9 +1,10 @@
-import { getResourceTemplate, findNode } from 'selectors/resourceSelectors'
+import { getResourceTemplate } from 'selectors/resourceSelectors'
 import _ from 'lodash'
 
 export default class Validator {
-  constructor(selectorReducer) {
+  constructor(selectorReducer, findNode) {
```

Comment:
Why pass findNode to Validator?

Reply:
This refactor enabled me to change the implementation easily.  Now that the refactor is done, we could back this out if you care about that.